searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.h

```diff
@@ -103,7 +105,14 @@ class WeakAndBlueprint : public IntermediateBlueprint
 
 //-----------------------------------------------------------------------------
 
-class NearBlueprint : public IntermediateBlueprint
+/** shared implementation for operators that degrade to AND when creating filter for upper-bound case */
+class FilterUpperAndBlueprint : public IntermediateBlueprint
```

Comment:
This is a slippery slope that could lead to templated and/or multiple inheritance real fast. I would prefer utility functions for creating AND/OR filter searches that are called from explicit implementations over this.

Reply:
Agree